src/transformers/integrations/bitsandbytes.py

```diff
@@ -95,7 +98,14 @@ class `Int8Params` from `bitsandbytes`.
             if is_8bit:
                 new_value = bnb.nn.Int8Params(new_value, requires_grad=False, **kwargs).to(device)
             elif is_4bit:
-                new_value = bnb.nn.Params4bit(new_value, requires_grad=False, **kwargs).to(device)
+                if new_value.dtype not in (torch.uint8,):
```

Comment:
```suggestion
                if new_value.dtype != torch.uint8:
```

Reply:
@younesbelkada ,
thank you for the supportive comments. I will make a major update once the bnb PR goes thru.
meanwhile, could you comment on my message [On storage format for the quantized parameters ](https://github.com/huggingface/transformers/pull/26037#issuecomment-1712438542)above? - I am not very comfortable with flooding the state dict with all components of quant_state as this PR does now.

Reply:
Done. `new_value.dtype != torch.uint8` suggestion accepted.